### ISSUE TITLE
False positive RedundantCondition and TypeDoesNotContainType error on explode

### ISSUE THREAD

**Issue by kkmuffme** [vimeo/psalm#2542]
There are 2 false positive errors with explode, which both are valid use cases though:

1) ERROR: RedundantCondition - 7:7 - Found a redundant condition when evaluating $exploded and trying to reconcile type 'non-empty-list<string>' to !empty

https://psalm.dev/r/4b2cba8724

```php
<?php
$string = 'hello,world';
$exploded = explode( ',', $string );

if ( !empty( $exploded ) ) {
    echo "foo";
}
```

2) ERROR: TypeDoesNotContainType - 7:6 - Found a contradiction when evaluating $exploded and trying to reconcile type 'non-empty-list<string>' to empty

https://psalm.dev/r/4c13b6a84c

```php
<?php
$string = 'hello world';
$exploded = explode( ',', $string );

if ( empty( $exploded ) ) {
  $exploded[] = 'foo';
}
```

**Comment by weirdan** [comment]
> There are 2 false positive errors with explode,

Why do you think they are false positives?

**Comment by muglug** [comment]
the result of `explode` when called with two arguments is never empty

**Comment by kkmuffme** [comment]
@muglug not true, the array may be empty, as in the example above.

empty() does check if an array is empty, so this is perfectly valid.

**Comment by muglug** [comment]
You might want to try running those examples on 3v4l.org :)

**Comment by kkmuffme** [comment]
Ah thx, really sorry, somehow got stuff mixed up (some after effects of new years :D)
